multi-user-dev-server.js

```diff
@@ -24,56 +24,38 @@ const CompilerCollection = require("./compiler-collection.js");
 function createDevServer(optionsFromUsername, expireUnusedAfterSeconds) {
   const app = Express();
 
-  // Map of username -> { compiler, watching, whenDone }
+  // Map of username -> { compiler, watching }
   const compilers = CompilerCollection(expireUnusedAfterSeconds);
 
   /**
    * Given a username of a user on the dev machine, this returns an object with:
    *   compiler: a webpack compiler instance
    *   watching: a watching instance from calling compiler.watch()
-   *   whenDone: returns a promise that resolves when the user's bundle is done
    *
    * @param string username A username, will be used in optionsFromUsername()
    *
-   * @return { compiler, watching, whenDone }
+   * @return { compiler, watching }
    */
   const getUserCompiler = username => {
     if (!/^[\w\-_]+$/.test(username)) {
       throw new Error('invalid characters in username');
     }
 
     const options = optionsFromUsername(username);
+    options.username = username;
```

Comment:
Wat? That seems weird. I would think that `optionsFromUsername` would add this value to the `options` object?

Reply:
Yes, but this function is outside the app (i.e. the mapping function is part of the input to the app) so placing an arbitrary restriction on it (that we could easily fulfill ourselves) seemed like a bad idea.

Reply:
Fair enough. That might be worth a comment since it looked odd to both of us.